src/AstTranslator.cpp

```diff
@@ -592,12 +608,15 @@ std::unique_ptr<RamOperation> AstTranslator::ProvenanceClauseTranslator::createO
             values.push_back(translator.translateValue(con->getLHS(), valueIndex));
             values.push_back(translator.translateValue(con->getRHS(), valueIndex));
         } else if (auto neg = dynamic_cast<AstProvenanceNegation*>(lit)) {
-            for (size_t i = 0; i < neg->getAtom()->getArguments().size() - 2; ++i) {
+            size_t numberOfHeights = getNumberOfHeights(neg->getAtom(), translator.program);
+
+            for (size_t i = 0; i < neg->getAtom()->getArguments().size() - 1 - numberOfHeights; ++i) {
```

Comment:
Yes, `neg->getAtom()->getArguments().size()` is always greater than or equal to `numberOfHeights + 1` because  `numberOfHeights + 1` new arguments were previously added to the atom during provenance transformation.